api/src/main/java/org/openmrs/module/coreapps/utils/VisitTypeHelper.java

```diff
@@ -0,0 +1,250 @@
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.JsonParseException;
+import org.codehaus.jackson.map.JsonMappingException;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.node.ArrayNode;
+import org.openmrs.Encounter;
+import org.openmrs.EncounterType;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.Person;
+import org.openmrs.Visit;
+import org.openmrs.VisitType;
+import org.openmrs.api.APIException;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.VisitService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.coreapps.CoreAppsConstants;
+import org.openmrs.module.emrapi.visit.VisitDomainWrapper;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Romain Buisson
+ *
+ */
+@Component
+public class VisitTypeHelper {
+
+	protected static final Log log = LogFactory.getLog(VisitTypeHelper.class);
+
+	/**
+	 * Returns a list of ordered visit types, provided by a global property
+	 * If a visit type is present in the visitTypes argument, but not found in the global property string,
+	 * it will be returned unordered, at the end of the visitTypesOrdered list
+	 *
+	 * @param visitTypes
+	 * @return visitTypesOrdered
+	 */
+	public List<VisitType> getOrderedVisitTypes (List<VisitType> visitTypes) {
+		AdministrationService adminService = Context.getAdministrationService();
+		String propertyValue = adminService.getGlobalProperty(CoreAppsConstants.VISIT_TYPES_ORDER_PROPERTY);
+		VisitService vs = Context.getVisitService();
+
+		return getOrderedVisitTypes(visitTypes, propertyValue, vs);
+	}
+
+	/**
+	 * Returns a list of ordered visit types.
+	 *
+	 * @param visitTypes All the visit types
+	 * @param propertyValue The visit types to order in JSON-like format
+	 * @param visitService
+	 * @return visitTypesOrdered The visit types ordered and merged with the input visit type list
+	 */
+	public List<VisitType> getOrderedVisitTypes (List<VisitType> visitTypes, String propertyValue,VisitService visitService) {
+
+		Map<Integer,String> order = null;
+		List<VisitType> visitTypesOrdered = new ArrayList<VisitType>();
+
+		if (propertyValue != null) {
+			try {
+				order = new ObjectMapper().readValue(propertyValue, HashMap.class);
+			} catch (JsonParseException e) {
+				VisitTypeHelper.log.error("Unable to parse global property \"" + CoreAppsConstants.VISIT_TYPES_ORDER_PROPERTY + "\"");
+				e.printStackTrace();
+			} catch (JsonMappingException e) {
+				VisitTypeHelper.log.error("Unable to map global property \"" + CoreAppsConstants.VISIT_TYPES_ORDER_PROPERTY + "\"");
+				e.printStackTrace();
+			} catch (APIException e) {
```

Comment:
It depends on what you do on handling each. My comment was based on how you have currently handled them. :)